src/app/src/components/EmbeddedMapTextSearchLabelConfig.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect } from 'react';
+import Typography from '@material-ui/core/Typography';
+import { withStyles } from '@material-ui/core/styles';
+import ControlledTextInput from './ControlledTextInput';
+import { DEFAULT_SEARCH_TEXT } from '../util/constants';
+
+const styles = {
+    subsectionHeader: {
+        color: 'rgb(0, 0, 0)',
+        fontSize: '15px',
+        fontWeight: 'bold',
+        margin: '10px 0',
+    },
+};
+
+function EmbeddedMapTextSearchLabelConfig({
+    textSearchLabel,
+    setTextSearchLabel,
+    anyFieldSearchable,
+}) {
+    const setTextToDefault = () => {
+        if (!textSearchLabel) {
+            setTextSearchLabel(DEFAULT_SEARCH_TEXT);
+        }
+    };
+
+    useEffect(() => {
+        if (!anyFieldSearchable) {
+            setTextSearchLabel(DEFAULT_SEARCH_TEXT);
+        }
+    }, [anyFieldSearchable]);
+
+    return (
+        <>
```

Comment:
This should probably be wrapped in a `<div>`, rather than a Fragment. That way it will be contained in a specific subtree of the DOM and can be considered a proper sibling of the other sub-sections (eg, Promoted Fields).

The wrapping `<div>` should have `marginTop: 20px` to match its siblings.